Robot-Tests/KW_Paallyste.robot

```diff
@@ -0,0 +1,119 @@
+#       Joni Laari      Sitowise Oy     2022
+
+*** Settings ***
+Documentation       Keywords for paving (Päällyste)
+Library    SeleniumLibrary
+Resource    common_keywords.robot
+
+*** Variables ***
+${LocatorForDDM}                            css=#feature-attributes .form-group.editable. select:first-of-type
+${LocatorForDDM_Selection}                  css=select >option:nth-child(2)
+${Rajoituksen_poisto}                       Haluatko varmasti poistaa rajoituksen?
+
+*** Keywords ***
+
+Päällyste_1   [arguments]     ${testipaikka}
+    Log     Päällysteen tarkastelu katselutilassa.
+    #testataan valinnan toimivuus
+    Vaihda Tietolaji                            ${TL_Päällyste_RB}
+    Paikanna osoite                             ${testipaikka}
+    Zoomaa kartta                               5   50 m
+    Odota sivun latautuminen
+    Click Element At Coordinates                ${Kartta}   0  20
+    Wait Until Element Is Visible               ${FA_otsikko}
+    
+    #Klikataan näkymä pois
+    Click Element At Coordinates                ${Kartta}  -100  -100
+
+    #double click
+    Tupla Klikkaa Kartan Keskelle
+    Wait Until Element Is Visible               ${FA_otsikko}
+
+
+Päällyste_2    [arguments]    ${testipaikka}    ${päällysteluokka}
+    Log                                         Testataan päällysteen vaihto
+    Testin Aloitus
+    Vaihda Tietolaji                            ${TL_Päällyste_RB}
+    Paikanna osoite                             ${testipaikka}
+    Zoomaa kartta                               5   50 m
+    Odota sivun latautuminen
+    Siirry Muokkaustilaan
+    Click Element At Coordinates                ${Kartta}   0   20
+
+    Wait Until Element Is Visible               css=.form-control.paved-road
+    Click Element                               css=.form-control.paved-road
+    Click Element                               ${päällysteluokka}
+    Wait Until Element Is Visible               ${FA_footer_Tallenna}
+
+    #muutoksen peruutus vs
+    Wait Until Element Is Visible               ${FA_footer_Peruuta}
+    Click Element                               ${FA_footer_Peruuta}
+
+    #muutoksen tallennuksen kokeileminen, otetaan käyttöön kun seleniumin klikkiongelmille viivakohteiden kanssa saadaan jokin selkeä syy tai korjaus    15.09.22
+    #Wait Until Element Is Visible               ${FA_footer_Tallenna}
+    #Click Element                               ${FA_footer_Tallenna}
+
+    Wait Until Element Is Not Visible           ${Spinner_Overlay}
```

Comment:
Näissä testeissä ois mun mielestä hyvä olla jotkin "should" lauseet, jotka tarkistaa, että lopputulos oli sellainen kuin haluttiin.

Reply:
Lisätty